npm/ipsm/ipsm_test.go

```diff
@@ -4,95 +4,97 @@ package ipsm
 
 import (
 	"fmt"
-	"log"
+	"io/ioutil"
 	"os"
+	"path/filepath"
 	"testing"
 
-	"github.com/Azure/azure-container-networking/npm/iptm"
 	"github.com/Azure/azure-container-networking/npm/metrics"
 	"github.com/Azure/azure-container-networking/npm/metrics/promutil"
 	"github.com/Azure/azure-container-networking/npm/util"
+	testingutils "github.com/Azure/azure-container-networking/test/utils"
 	testutils "github.com/Azure/azure-container-networking/test/utils"
 	"github.com/stretchr/testify/require"
-	"k8s.io/utils/exec"
 )
 
 func TestSave(t *testing.T) {
-	ipsMgr := NewIpsetManager(exec.New())
-	if err := ipsMgr.Save(util.IpsetTestConfigFile); err != nil {
-		t.Errorf("TestSave failed @ ipsMgr.Save")
+	var calls = []testutils.TestCmd{
+		{Cmd: []string{"ipset", "save", "-file", "ipset.conf"}},
 	}
+
+	fexec := testutils.GetFakeExecWithScripts(calls)
+	ipsMgr := NewIpsetManager(fexec)
+	defer testingutils.VerifyCalls(t, fexec, calls)
+	err := ipsMgr.Save("ipset.conf")
+	require.NoError(t, err)
 }
 
 func TestRestore(t *testing.T) {
-	ipsMgr := NewIpsetManager(exec.New())
+	tmpFile, err := ioutil.TempFile(os.TempDir(), filepath.Base(util.IpsetTestConfigFile))
```

Comment:
Prefer to having a brief comment about filepath.Base like only using last element in path, but no strong opinion.